event-sourcing/event-repository/event-repository-jdbc/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/eventstream/EventStreamJdbcRepositoryIT.java

```diff
@@ -0,0 +1,260 @@
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+        final List<EventStream> streamOfStreams = jdbcRepository.next(3L, 2).collect(toList());
+
+        assertThat(streamOfStreams, hasSize(2));
+        assertThat(streamOfStreams.get(0).getStreamId(), is(streamId3));
+        assertThat(streamOfStreams.get(0).getSequenceNumber(), is(3L));
+        assertThat(streamOfStreams.get(1).getStreamId(), is(streamId2));
+        assertThat(streamOfStreams.get(1).getSequenceNumber(), is(2L));
+    }
+
+    @Test
+    public void shouldReturnPreviousRecords() throws InvalidSequenceIdException, SQLException {
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+        final UUID streamId4 = randomUUID();
+        final UUID streamId5 = randomUUID();
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+        final List<EventStream> streamOfStreams = jdbcRepository.previous(3L, 2).collect(toList());
+        assertThat(streamOfStreams, hasSize(2));
+
+        assertThat(streamOfStreams.get(0).getStreamId(), is(streamId4));
+        assertThat(streamOfStreams.get(0).getSequenceNumber(), is(4L));
+
+        assertThat(streamOfStreams.get(1).getStreamId(), is(streamId3));
+        assertThat(streamOfStreams.get(1).getSequenceNumber(), is(3L));
+    }
+
+
+    @Test
+    public void shouldReturnFirstRecords() throws InvalidSequenceIdException, SQLException {
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+        final UUID streamId4 = randomUUID();
+        final UUID streamId5 = randomUUID();
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
```

Comment:
again, don't think we need this many events. 3 or 4 would prove the point.

Reply:
shorted for just 3 events